cdap-app-templates/cdap-etl/cdap-etl-realtime/src/test/java/co/cask/cdap/template/etl/realtime/RealtimeCubeSinkTest.java

```diff
@@ -108,11 +110,22 @@ public void test() throws Exception {
 
     AdapterManager manager = createAdapter(adapterId, adapterConfig);
 
-    long startTs = System.currentTimeMillis() / 1000;
+    final long startTs = System.currentTimeMillis() / 1000;
 
     manager.start();
-    // Let the worker run for 5 seconds
-    TimeUnit.SECONDS.sleep(5);
+    Tasks.waitFor(true, new Callable<Boolean>() {
+      @Override
+      public Boolean call() throws Exception {
+        long endTs = System.currentTimeMillis() / 1000;
+        DataSetManager<Cube> tableManager = getDataset("cube1");
+        Cube cube = tableManager.get();
+        Collection<TimeSeries> result = cube.query(CubeQuery.builder()
```

Comment:
No, because the query includes a variable, endTs, the value of which changes between this line and line 136.